pkg/realizer/healthcheck/healthcheck.go

```diff
@@ -75,25 +77,117 @@ func DetermineHealthCondition(rule *v1alpha1.HealthRule, realizedResource *v1alp
-				} else {
-					return conditions.SingleConditionMatchCondition(metav1.ConditionUnknown, rule.SingleConditionType)
-				}
+			if rule.MultiMatch != nil {
+				return multiMatchCondition(rule.MultiMatch, stampedObject)
 			}
-			return conditions.SingleConditionMatchCondition(metav1.ConditionUnknown, "")
 		}
 	}
 	return conditions.UnknownResourcesHealthyCondition()
 }
+
+func singleConditionTypeCondition(singleConditionType string, stampedObject *unstructured.Unstructured) metav1.Condition {
+	singleCondition := extractConditions(stampedObject).ConditionWithType(singleConditionType)
+	if singleCondition != nil {
+		if singleCondition.Status == metav1.ConditionFalse || singleCondition.Status == metav1.ConditionTrue {
+			return conditions.SingleConditionMatchCondition(singleCondition.Status, singleConditionType, singleCondition.Message)
+		} else {
+			return conditions.SingleConditionMatchCondition(metav1.ConditionUnknown, singleConditionType, singleCondition.Message)
+		}
+	}
+	return conditions.SingleConditionMatchCondition(metav1.ConditionUnknown, singleConditionType, fmt.Sprintf("condition with type [%s] not found on resource status", singleConditionType))
+}
+
+func multiMatchCondition(multiMatchRule *v1alpha1.MultiMatchHealthRule, stampedObject *unstructured.Unstructured) metav1.Condition {
+	condition := anyUnhealthyMatchCondition(multiMatchRule.Unhealthy, stampedObject)
+	if condition != nil {
+		return *condition
+	}
+	condition = allHealthyMatchCondition(multiMatchRule.Healthy, stampedObject)
+	if condition != nil {
+		return *condition
+	}
+	return conditions.MultiMatchNoMatchesCondition()
+}
+
+func extractConditions(stampedObject *unstructured.Unstructured) conditions.ConditionList {
+	var conditionList conditions.ConditionList
+	maybeStatus := stampedObject.UnstructuredContent()["status"]
+	if unstructuredStatus, statusOk := maybeStatus.(map[string]interface{}); statusOk {
+		maybeConditions := unstructuredStatus["conditions"]
+		maybeConditionsJSON, err := json.Marshal(maybeConditions)
+		if err == nil {
+			err = json.Unmarshal(maybeConditionsJSON, &conditionList)
+			if err != nil {
+				return conditions.ConditionList{}
+			}
+		}
+	}
+	return conditionList
+}
+
+func messageForMatchingFieldRequirement(requirement v1alpha1.HealthMatchFieldSelectorRequirement, stampedObject *unstructured.Unstructured) string {
+	evaluator := eval.EvaluatorBuilder()
+	fieldValue, fieldErr := evaluator.EvaluateJsonPath(requirement.Key, stampedObject.UnstructuredContent())
+	if fieldErr != nil {
+		fieldValue = "<error retrieving field value>"
+	}
+	messageValue, messageErr := evaluator.EvaluateJsonPath(requirement.MessagePath, stampedObject.UnstructuredContent())
+	if messageErr != nil {
+		messageValue = fmt.Sprintf("unknown, error retrieving message path [%s]", requirement.MessagePath)
+	}
+
+	return fmt.Sprintf("field value: %v, message: %v", fieldValue, messageValue)
```

Comment:
I'm not sure if once this bubble up to the owners' statuses this will be enough to be helpful in failure scenario 🤔  e.g., looking at the kuttl tests (`supply-chain/workload-status/`) where a configmap is healthy according to

```yaml
  healthRule:
    multiMatch:
      healthy:
        matchFields:
          - key: .data.favorite_color
            operator: Exists
            messagePath: .metadata.name
      unhealthy:
        matchFields:
          - key: .data.kill_switch
            operator: Exists
            messagePath: .metadata.name
```

we get 

```yaml
      conditions:
        - type: ResourceSubmitted
          status: "True"
          reason: ResourceSubmissionComplete
        - type: Healthy
          status: "True"
          reason: MatchedField
          message: "field value: knitting, message: example-config-configmap"
```

which makes me wonder if we should some how include the `key` that's been used for that lookup 🤔  something along the lines of 

```yaml
      conditions:
        - type: ResourceSubmitted
          status: "True"
          reason: ResourceSubmissionComplete
        - type: Healthy
          status: "True"
          reason: MatchedField
          message: "key: '.data.favorite_color', field value: 'knitting', message: 'example-config-configmap'"
```

wdyt?

Reply:
^ although .. sometimes `key` can be long

```
- key: 'status.conditions[?(@.type=="ReconcileFailed")].status'
```

or similar